### DISCUSSION TITLE
PNG to sprite for nearly transparent background...Will this be a problem?

### DISCUSSION THREAD

**Discussion by MYCAMEL222** [Bodmer/TFT_eSPI#2685] [Q&A - Sprites] [answered]
I am using Flash Arrays to store PNG icons. I Decode them to from a 50x50 PNG to a 50x50 sprite and push the sprite with the background color of the PNG as transparent. If I create the PNG with a background color close to the color of what I am pushing the sprite over, you really cant tell the PNG is not transparent.

My question is...is Decoding PNG to a Sprite the correct way to do this....?

See code snippet below.

img.createSprite(50,50);  
  rc = png.openFLASH((uint8_t *)pausegray, sizeof(pausegray), pngDrawSprite);
  if (rc == PNG_SUCCESS) {
    img.startWrite();
    uint32_t dt = millis();
    rc = png.decode(NULL, 0);
    img.endWrite();
  }
     img.pushSprite(70,180,0xDEDB);
  img.deleteSprite();
}
void pngDrawSprite(PNGDRAW *pDraw) {
  uint16_t lineBuffer[MAX_IMAGE_WDITH];
  png.getLineAsRGB565(pDraw, lineBuffer, PNG_RGB565_BIG_ENDIAN, 0xffffffff);
  img.pushImage(xpos, ypos + pDraw->y, pDraw->iWidth, 1, lineBuffer);
}

![Untitled design (52)](https://github.com/Bodmer/TFT_eSPI/assets/85456085/09cba9f2-94c5-4573-9689-bc9fe0eeb23a)
![20230602_130903](https://github.com/Bodmer/TFT_eSPI/assets/85456085/854de956-2a09-4d5e-ba4c-115e9d13184e)

**Comment by Bodmer** [comment, marked as answer]
PNG images can have inbuilt transparency. See exampe here:
https://github.com/Bodmer/TFT_eSPI/tree/master/examples/PNG%20Images/Flash_transparent_PNG

**Reply by MYCAMEL222** [reply]
That works much better...

Somehow, I had stripped down the pngDEC code to not pick up the transparency mask.

I pasted the code from the example and saved one of my PNG icons with transparency. 

Works like a champ!!!

Thanks for another lesson. I'll stop buggin you.

**Reply by Bodmer** [reply]
No problem! Transparent PNG images to sprites is not supported yet but there is example code here if you need it:

https://github.com/Bodmer/TFT_eSPI/discussions/2660
